tests/aws/services/sns/test_sns.py

```diff
@@ -3810,7 +3811,14 @@ def _clean_headers(response_headers: dict):
             snapshot.match("http-message", payload)
             snapshot.match("http-message-headers", _clean_headers(notification_request.headers))
 
-        unsub_request = requests.get(expected_unsubscribe_url)
+        auth_header = aws_stack.mock_aws_request_headers(
+            "sns", region_name=TEST_AWS_REGION_NAME, access_key=TEST_AWS_ACCESS_KEY_ID
+        )["Authorization"]
+
+        unsub_request = requests.get(
+            url=expected_unsubscribe_url, headers={"Authorization": auth_header}
+        )
+
```

Comment:
For posterity, authentication params are required when calling API via HTTP <https://docs.aws.amazon.com/sns/latest/api/API_Unsubscribe.html#API_Unsubscribe_Examples>. `AUTHPARAMS` is a SigV4 signature that can be sent either via headers (as is done here) or as query params <https://docs.aws.amazon.com/IAM/latest/UserGuide/aws-signing-authentication-methods.html>

@sannya-singal could you please verify that use of `mock_aws_request_headers` does not break the test when running against AWS?